js/common/modules/@arangodb/foxx/check-args.js

```diff
@@ -207,64 +242,90 @@ exports.validateStatus = function (value) {
   return {value: status, error: null};
 };
 
-exports.validateSchema = function (value) {
-  if (value) {
-    if (value.isJoi || typeof value.validate === 'function') {
-      return {value, error: null};
+exports.validateSchema = function (schema = true) {
+  if (schema === null) {
+    return {
+      value: {schema: false, validate: createSchemaValidator(false)},
+      error: null
+    };
+  }
+  if (isJoiSchema(schema)) {
+    if (schema.isJoi) {
+      return {
+        value: {schema, validate: createSchemaValidator(schema)},
+        error: null
+      };
     }
     try {
-      return {value: joi.object(value).required(), error: null};
+      schema = joi.object(schema).required();
```

Comment:
For legacy reasons we allow passing "sloppy" JOI schemas that are plain JS objects where every property value is a JOI schema. Our `isJoiSchema` helper detects these as JOI schemas but we need to convert them. For ArangoDB 4 we probably don't want to continue supporting these so we can drop JOI.